service/src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java

```diff
@@ -111,41 +113,104 @@ public void setUp() throws Exception {
+    }
+
+
     @Test
     public void testSemanticsFailureEvent() throws Exception {
-        ResponseEntity<?> elem = unit.generate("eiffelsemantics", "EiffelActivityFinished", false, false, true, 1, false, body.getAsJsonObject());
+        File file = new File("src/test/resources/ErlookupConfidenceLevelOutput.json");
+        JsonParser parser = new JsonParser();
+        JsonObject json = parser.parse(new FileReader(file)).getAsJsonObject();
+        ResponseEntity<?> elem = unit.generate("eiffelsemantics", "EiffelActivityFinished", false, false, true, 1, false, json);
+        assertEquals(elem.getStatusCode(), HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testSemanticsFailureEventArray() throws Exception {
+        File file = new File("src/test/resources/ErlookupConfidenceLevelArrayOutput.json");
+        JsonParser parser = new JsonParser();
+        JsonArray json = parser.parse(new FileReader(file)).getAsJsonArray();
+        ResponseEntity<?> elem = unit.generate("eiffelsemantics", "EiffelActivityFinished", false, false, true, 1, false, json);
         assertEquals(elem.getStatusCode(), HttpStatus.BAD_REQUEST);
     }
 
     @Test
     public void testEiffel3SuccessEvent() throws Exception {
-        ResponseEntity<?> elem = unit.generate("eiffel3", "eiffelartifactnew", false, false, true, 1, false, body.getAsJsonObject());
+        File file = new File("src/test/resources/ErlookupConfidenceLevelOutput.json");
+        JsonParser parser = new JsonParser();
+        JsonObject json = parser.parse(new FileReader(file)).getAsJsonObject();
+        ResponseEntity<?> elem = unit.generate("eiffel3", "eiffelartifactnew", false, false, true, 1, false, json);
+        assertEquals(elem.getStatusCode(), HttpStatus.OK);
+    }
+
+    @Test
+    public void testEiffel3SuccessEventArray() throws Exception {
+        File file = new File("src/test/resources/ErlookupConfidenceLevelArrayOutput.json");
+        JsonParser parser = new JsonParser();
+        JsonArray json = parser.parse(new FileReader(file)).getAsJsonArray();
+        ResponseEntity<?> elem = unit.generate("eiffel3", "eiffelartifactnew", false, false, true, 1, false, json);
         assertEquals(elem.getStatusCode(), HttpStatus.OK);
     }
 
     @Test
     public void testEiffel3FailureEvent() throws Exception {
-        ResponseEntity<?> elem = unit.generate("eiffel3", "eiffelartifactnewevent", false, false, true, 1, false, body.getAsJsonObject());
+        File file = new File("src/test/resources/ErlookupConfidenceLevelOutput.json");
+        JsonParser parser = new JsonParser();
+        JsonObject json = parser.parse(new FileReader(file)).getAsJsonObject();
+        ResponseEntity<?> elem = unit.generate("eiffel3", "eiffelartifactnewevent", false, false, true, 1, false, json);
         assertEquals(elem.getStatusCode(), HttpStatus.BAD_REQUEST);
     }
 
+    @Test
+    public void testEiffel3FailureEventArray() throws Exception {
```

Comment:
Does the test pass? Real events are passed to generate method instead of templates...